packages/contractkit/src/identity/offchain/schema-utils.ts

```diff
@@ -21,62 +29,399 @@ export class OffchainError extends RootError<SchemaErrorTypes.OffchainError> {
   }
 }
 
-type SchemaErrors = InvalidDataError | OffchainError
+export class UnknownCiphertext extends RootError<SchemaErrorTypes.UnknownCiphertext> {
+  constructor() {
+    super(SchemaErrorTypes.UnknownCiphertext)
+  }
+}
+
+export class UnavailableKey extends RootError<SchemaErrorTypes.UnavailableKey> {
+  constructor() {
+    super(SchemaErrorTypes.UnavailableKey)
+  }
+}
 
-export class SingleSchema<T> {
+type SchemaErrors = InvalidDataError | OffchainError | UnknownCiphertext | UnavailableKey
+
+const ioTsToSolidityTypeMapping: { [x: string]: string } = {
+  [t.string._tag]: 'string',
+  [t.number._tag]: 'uint256',
+  [t.boolean._tag]: 'bool',
+}
+export type EIP712Schema = Array<{ name: string; type: string }>
+const binaryEIP712Schema: EIP712Schema = [
+  { name: 'path', type: 'string' },
+  { name: 'hash', type: 'string' },
+]
+
+export class SimpleSchema<DataType> {
```

Comment:
Yeah, we do specifically want signatures to include the path though so that they can't be shuffled around. I personally am not worried about "constructing a new schema".